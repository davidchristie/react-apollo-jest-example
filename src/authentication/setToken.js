import TOKEN_KEY from './TOKEN_KEY'

export default function setToken (token) {
  window.localStorage.setItem(TOKEN_KEY, token)
}
