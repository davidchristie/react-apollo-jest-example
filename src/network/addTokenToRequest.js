import getToken from '../authentication/getToken'

export default function addTokenToRequest (request) {
  const token = getToken()
  if (token) {
    if (!request.options.headers) {
      request.options.headers = {}
    }
    request.options.headers.Authorization = `Bearer ${token}`
  }
}
