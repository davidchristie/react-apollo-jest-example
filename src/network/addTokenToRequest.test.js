import setToken from '../authentication/setToken'
import addTokenToRequest from './addTokenToRequest'

it('does nothing', () => {
  const request = {
    options: {}
  }
  addTokenToRequest(request)
  expect(request).toMatchSnapshot()
})

it('adds token to request headers', () => {
  setToken('token')
  const request = {
    options: {
      headers: {}
    }
  }
  addTokenToRequest(request)
  expect(request).toMatchSnapshot()
})

it('adds headers and token to request', () => {
  setToken('token')
  const request = {
    options: {}
  }
  addTokenToRequest(request)
  expect(request).toMatchSnapshot()
})
