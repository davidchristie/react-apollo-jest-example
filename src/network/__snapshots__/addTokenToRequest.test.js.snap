// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds headers and token to request 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "Authorization": "Bearer token",
    },
  },
}
`;

exports[`adds token to request headers 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "Authorization": "Bearer token",
    },
  },
}
`;

exports[`does nothing 1`] = `
Object {
  "options": Object {},
}
`;
